src/NHibernate/Persister/Entity/SingleTableEntityPersister.cs

```diff
@@ -161,7 +161,14 @@ public class SingleTableEntityPersister : AbstractEntityPersister, IQueryable
 				constraintOrderedKeyColumnNames[position] = keyColumnNames[i];
 			}
 
-			spaces = qualifiedTableNames.Concat(persistentClass.SynchronizedTables).ToArray();
+			if (persistentClass.HasUpdateTimestampsCache)
+			{
+				spaces = qualifiedTableNames.Concat(persistentClass.SynchronizedTables).ToArray();
+			}
+			else
+			{
+				spaces = persistentClass.SynchronizedTables.ToArray();
```

Comment:
Because of this modification AutoFlush "ShouldAutoFlushWhenInExplicitTransactionAsync" test fails if "Entity" never cached?
```c#
rc.Cache(x => x.Usage(CacheUsage.Never))
```
Whats is the best way  not update never cached entity UpdateTimestampsCache?
Maybe ActionQueue I could check everytime executedSpaces, with an helper**(?)** method retrieve entity persister from tableName and if persister HasUpdateTimestampsCache = true then don't add executedSpaces?

Main issue is QuerySpaces is tablenames, but HasUpdateTimestampsCache option is entity persister map, how to get entity persister from table names at ActionQueue?

Or basicly store and use UpdateTimestampsSpaces at persistent class?